opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1000,21 +1001,32 @@ namespace Opm {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
```

Comment:
Hmmm... If that does not work it would be quite serious -- it is the main way to interact with a dune grid these days I believe, and the partition stuff is the main way to deal with parallelism in the interface. Could you give it a shot?